### ISSUE TITLE
Novas Tabelas Correios - Capital e Interior

### ISSUE THREAD

**Issue by Cellin** [pedro-teixeira/correios#341]
Olá a todos!

Como vocês já devem estar sabendo, dia 06/03/2018 o Correios vai liberar uma nova tabela, com novos preços, e algumas regiões com saída fora de São Paulo vão ficar inviáveis, pelo menos no meu caso.

Para o interior do centro-oeste (exceto MS), norte e nordeste, os valores já começam a ficar fora da realidade das coisas... Acima de R$ 50,00, chegando até R$ 82,00, e detalhe, para até 300 gramas, independente do tamanho do teu pacote e do espaço que ele ocupe no caminhão.

Muitos vão ter que parar de vender para estes locais (eu sou um deles, com saída de SC), queria saber a opinião de vocês e qual soluções estão pensando? 

No meu caso, ainda não pensei em nada, mas, vou precisar remover a disponibilidade de entrega no interior destas regiões (centro-oeste, norte e nordeste).

Abraços,

**Comment by rafaelpatro** [comment]
Então @Cellin, tb recebi essa carta.

Os itens que mais me preocuparam foram esses...

 * Ajuste na precificação de encomendas não maquináveis, cujas embalagens não sejam retangulares ou quadradas. Para elas será acrescida uma cobrança adicional “Outros formatos” no valor de R$20,00.

 * Diminuição do mínimo para mensuração do peso cúbico, de 10kg para 5kg.

O último item parece bem claro, apesar de estrategicamente aumentar o preço, sem que haja muita repercussão. Por exemplo, quem trabalha com objetos leves maiores que 31 cm em todos os lados, terá um impacto financeiro negativo bem maior.
Ex.: 31x31x31/6000 ~= 5kg

Quanto ao ajuste de encomendas (não maquináveis) não retangulares... não sei exatamente oq significa o termo _não maquinável_. Espero estar errado, mas Imagino que rolos, cilindros, pacotes (não retangulares) terão seus valores acrescidos de 20 reais. E isso deve impactar muitas empresas tb.

Tudo isso parece uma estratégia de minar a empresa, afastar os clientes, e finalmente vender a única estatal que realmente cobre todo o território nacional.

**Comment by redbio** [comment]
@rafaelpatro  Mas isso não afeta o calculo do módulo ou é transparente?

Na carta que recebi ainda aparece essa informação:

- Devido ao elevado número de roubo de carga, será incorporada a cobrança emergencial de R$ 3,00 (três reais) ao valor da postagem para encomendas destinadas, neste momento, ao Rio de Janeiro e região metropolitana;

**Comment by rafaelpatro** [comment]
Não sei qto a taxa de 3 reais, mas qto a mensuração do peso cubado, o módulo vai continuar funcionando normalmente. Só limpar a cache no dia 6.

**Comment by KARINABORGES** [comment]
Olá pessoal, tenho uma loja virtual pelo magento e após os reajustes de preços dos fretes, o módulo que eu usava passou a dar problemas pois não estava calculando pelas medidas das caixas e consequentemente não estava saindo certo os valores. Meu técnico mudou para o módulo do Pedro Teixeira, porém também não está saindo valores corretos. No meu caso, eu tenho produtos de diversos tamanhos que geralmente são comprados juntos, portanto preciso de um módulo que considere a medida da caixa do maior produto e some apenas o peso dos produtos. Porém o do Pedro Teixeira parece que está somando os pesos e também as medidas, aí sai um valor absurdo. Não sei mais o que fazer pois não conseguimos descobrir como resolver e estou perdendo muito dinheiro com isso devido ao reajuste abusivo dos correios. Principalmente porque a maioria dos meus produtos tem caixa maior que 42cm. Por favor será que alguém pode me ajudar? Não sei mais onde recorrer. achei esse site que nem sei se é pra isso que serve, mas como vi vocês comentando sobre isso acredito que podem me ajudar. Muito obrigada!

**Comment by rafaelpatro** [comment]
@KARINABORGES, o módulo do pedro dimensiona cada produto, e soma o total no final da conta. Ele trabalha com os valores cadastrados, e a margem de erro é bem pequena, se comparado a outros módulos que nem calculam a cubagem.

Se vc tem algum item que pode ser encaixado, e economizar espaço (ex.: balde). Pode configurar o atributo _Diferença do Encaixe_. Nesse caso levando 2 unidades o módulo soma apenas a diferença.
Isso serve apenas para quantidades maiores de itens únicos.

**Comment by KARINABORGES** [comment]
Obrigada por responder Rafael! Sim eu ja cadastrei essa diferença de
encaixe. Mas no caso de produtos diferentes que eu coloco na mesma caixa
como posso resolver? Por exemplo: eu vendo peças de alto falante, sendo que
tem cones de 12 polegadas (com medida alt 10xcomp 30x larg 30) e as vezes
no mesmo pedido o cliente compra um cone de 15" (15x42x42). Nesse exemplo o
frete está ficando muito caro, se eu deixo só o cone de 15 dá certinho ele
calcula a cubagem ok. Mas quando acrescentamos outro produto como cone de
12 aí o frete aumenta muito. Sendo que na realidade eles vão na mesma cx de
15 e o peso não vai mudar quase nada. Por favor me ajude não sei mais onde
recorrer. Obrigada

Em Qua, 28 de mar de 2018 17:41, Rafael Patro <notifications@github.com>
escreveu:

**Comment by rafaelpatro** [comment]
Bom dia
Infelizmente não há uma solução pronta para tratar seu caso. Cada empresa terá suas próprias regras.

Uma sugestão de melhoria para futuras implementações seria criar um sistema de regras no módulo.

O módulo não tem como adivinhar essas regras. Então seria interessante desenvolvermos algo afim de montar regras próprias de cada produto.

Ex.: Se o produto X e o produto Y estiverem no carrinho, reduzir o volume em 40%.

Enfim algo nesse sentido.

**Comment by KARINABORGES** [comment]
Rafael obrigada por sua atenção! Eu estava usando um outro módulo
"correios" que estava atendendo perfeitamente porém com essa nova regra de
cubagem ele começou a calcular errado e meu técnico não soube como corrigir
então tentou mudar. Agora estou vendendo com valores de fretes errados o
que está impactando diretamente no meu faturamento. Você conhece algum
outro módulo que poderia me atender nesse caso?

Em Qui, 29 de mar de 2018 08:36, Rafael Patro <notifications@github.com>
escreveu:

**Comment by rafaelpatro** [comment]
Não conheço outro módulo q calcule a cubagem melhor do q esse.

**Comment by KARINABORGES** [comment]
Rafael desculpe-me a insistência, mas realmente estou sem saída. Não consigo acreditar que um módulo desse porte não tenha pensado na possibilidade de ter mais de um modelo de produto comprado num mesmo pedido e que serão embalados juntos. Creio que não sou a única a passar por esta dificuldade. Dentro do seu vasto conhecimento nessa linguagem de programação, você realmente não consegue pensar em nada que possa me ajudar? Eu diminuí um pouco as medidas dos produtos e quando eles se unem aparentemente dá certo, porém quando aumento as quantidades volta a dar divergência no frete ou quando acrescento um produto diferente. Por favor não desista de mim :(

 

Atc.

Karina Borges

 <http://www.coimbracones.com.br> www.coimbracones.com.br

(35) 3422-2226 ou 98881-4773 whats



 

De: Rafael Patro [mailto:notifications@github.com] 
Enviada em: quinta-feira, 29 de março de 2018 09:39
Para: pedro-teixeira/correios
Cc: KARINABORGES; Mention
Assunto: Re: [pedro-teixeira/correios] Novas Tabelas Correios - Capital e Interior (#341)

 

Não conheço outro módulo q calcule a cubagem melhor do q esse.

**Comment by rafaelpatro** [comment]
Sim, também tenho interesse na regra que vc pediu. Espero trabalhar nisso em breve. Mas não tenho previsão.
Quem dera eu tivesse tempo de implementar tudo oq eu preciso.

Mas se não encontrou a solução que procura, talvez deva repensar, e adaptar sua necessidade às soluções disponíveis. Exemplo: Pode criar um kit de produtos no Magento, com os dois itens que descreveu. Então vc ajusta o peso e tamanho de forma otimizada. Na página de cada item independente, vc pode exibir um link do tipo "Compre junto" ou algo similar.

**Comment by redbio** [comment]
@KARINABORGES Concordo com o pessoal esse módulo é o melhor que tem, calcula a cubagem corretamente. Seu caso é bem específico, também temos esses problemas. Pense assim, você precisa cadastrar uma caixa master, deverias usar na própria configuração do módulo, onde se cadastra dimensões padrões. Criar uma propriedade no produto onde você indica que: usar caixa master se outro produto estiver no carrinho (tipo um check box). Aí seu técnico calcula quantos itens do carrinho cabe nessa caixa master e que possuem esse check box, aí segue o fluxo do cálculo, do contrario considera a dimensão de cada produto. Simples.


Bom existem várias formas de fazer, entendo que montar kits é uma opção.

**Comment by KARINABORGES** [comment]
@redbio obrigada por compartilhar sua experiência. Você disse que também tem esse problema, poderia me passar seu contato para maiores informações e se for possível me prestar este serviço. Pois meu técnico está impossibilitado de me ajudar neste caso. Aguardo seu retorno, muito obrigada!

**Comment by redbio** [comment]
@KARINABORGES  Oi tudo bem? Então no meu caso, na minha loja, eu fiz um padrão de caixas e na verdade não vejo o resultado de envio individualmente, ou seja, tem uns envios que a diferença é de centavos e outros que eu acabo ganhando, então eu olho para o todo no fim do mês e um envio cobre o prejuízo do outro. Acabei nem precisando mudar o modulo.
Mas eu posso tentar te ajudar, também está bem corrido aqui, bom se quiser segue me face: https://www.facebook.com/allan.b.santos   ou na loja pode pegar e me chamar no whats. www.mdagames.com.br

**Comment by KARINABORGES** [comment]
@redbio Allan boa tarde, tentei te ligar no whats que tem no seu site mas não atendeu. Tem algum outro contato? vou te deixar uma msg lá aí assim que você puder me retornar eu agradeço.

**Comment by rafaelpatro** [comment]
E a saga continua...
[Com ação coletiva, ABComm consegue brecar reajuste abusivo de serviços dos Correios](https://ecommercenews.com.br/noticias/lancamentos/com-acao-coletiva-abcomm-consegue-brecar-reajuste-abusivo-de-servicos-dos-correios/)

**Comment by KARINABORGES** [comment]
Nossa que notícia maravilhosa! se isso acontecer, será a solução para os meus problemas! obrigada @rafaelpatro por compartilhar...
